DragonSlayer.cpp

```diff
@@ -1,16 +1,32 @@
 #include "DragonSlayer.h"
 #include "Dragon.h"
+#include "Utility.h"
 
 //DragonSlayer::DragonSlayer
+DragonSlayer::DragonSlayer(const std::string name_, int hp, int armor_) :
+Character( hp, armor_, 4),
+attackItem(new AttackItem),
+name(name_)
+{
+    helpfulItems = makeHelpfulItems(2);
+    defensiveItems = makeDefensiveItems(1);
+}
 
 //DragonSlayer::getName
+const std::string& DragonSlayer::getName() {return name;}
+
+
 
 void DragonSlayer::attack(Character& other)
 {
     std::cout << name << " is attacking " << other.getName() << " !!" << std::endl;
     if( auto* dragon = dynamic_cast<Dragon*>(&other) )
     {
-        assert(false);
+        if (dragon->getHP() > 0 && attackItem != nullptr)
+        {
+            attackItem->use(this);
+        }
```

Comment:
This item is one-time use only.
you should reset it after using it.
```
        //note: items are single-use only, so you need to reset it after use.  
        //look in the Character class for how the other item types are reset after use.
```
- [ ] revise